spec/services/pdf_parser_spec.rb

```diff
@@ -5,15 +5,30 @@
 include AuthenticationSpecHelpers
 
 RSpec.describe PdfParser do
-  context 'when the PDF file is valid' do
-    let(:pdf_file) { Rack::Test::UploadedFile.new('spec/fixtures/sample.pdf', 'application/pdf') }
-    let!(:user) { authenticated }
+  let(:pdf_file) { Rack::Test::UploadedFile.new('spec/fixtures/sample.pdf', 'application/pdf') }
+  let!(:user) { authenticated }
+  let(:pdf_parser) { double(PdfParser) }
```

Comment:
We shouldn't have a double for the object under test. You can use a double to represent something that you don't have control over, or is outside the scope of the test, but we need to be testing the real PdfParser.